source/common/http/header_map_impl.cc

```diff
@@ -368,8 +379,10 @@ void HeaderMapImpl::insertByKey(HeaderString&& key, HeaderString&& value) {
     if (*lookup.value().entry_ == nullptr) {
       maybeCreateInline(lookup.value().entry_, *lookup.value().key_, std::move(value));
     } else {
+      auto delimiter =
+          delimiterByHeader(*lookup.value().key_, header_map_correctly_coalesce_cookies_);
       const uint64_t added_size =
-          appendToHeader((*lookup.value().entry_)->value(), value.getStringView());
+          appendToHeader((*lookup.value().entry_)->value(), value.getStringView(), delimiter);
```

Comment:
hm, I wonder if given the prior fix was insufficient, if we should just determine the delimiter in appendToHeader rather than require each call site to pass it in.  WDYT?

Reply:
I agree. 👍

Reply:
oh, It's look like we cannot determine the delimiter in `appendToHeader`. `apendInline` and some other methods may use a user-defined delimiter. If we add additional logic in the `appendToHeader`, it may prevent `apendInline` from working properly.